### ISSUE TITLE
DocBook 5 Validation [sf#247]

### ISSUE THREAD

**Issue by tomschr** [openSUSE/daps#246]
**Reported by tom_schr on 2014-10-30 15:41 UTC**
(Taken over from discussion [f3354b7f])

In 2012 an [issue regarding validation of DocBook 5](https://sourceforge.net/p/daps/tickets/97/) had been resolved, but with current DAPS 2 (trunk [r4350]) it was actually not possible for me to validate any DocBook 5 XML source.

It turned out that the validate.mk passes a variable `DOCBOOK5_RNG_URI` to the validator (`jing`), which is nowhere defined.
I could work around this by just adding

```
DOCBOOK5_RNG_URI=/usr/share/xml/docbook/schema/rng/5.0/docbook.rnc>
```

at the top of `validate.mk`.

Also the daps XSLT wrapper script under

```
/usr/share/daps/libexec/daps-xslt
```

does not work on my openSUSE because it calls `/usr/bin/saxon` which does not exist.
Workaround:

```
cd /usr/bin; ln -s saxon6 saxon
```

**Comment by tomschr** [comment]
**Updated by fsundermeyer on 2015-01-26 08:39**
- **status**: open --> invalid

**Comment by tomschr** [comment]
**Commented by fsundermeyer on 2015-01-26 08:39**

> It turned out that the validate.mk passes a variable `DOCBOOK5_RNG_URI` to > the validator (`jing`), which is nowhere defined.

This is not true. DOCBOOK5_RNG_URI is defined in /etc/daps/config:

```
## Key: DOCBOOK5_RNG_URI        
## ------------------
## Description: URI to DocBook 5 Relax NG schema
## Type:        URI
## Default:     "http://docbook.org/xml/5.0/rng/docbook.rng"
#
# URI to the DocBook 5 Relax NG schema that can be resolved by xmlcatalog
# There should be no need to change this entry
#
DOCBOOK5_RNG_URI="http://docbook.org/xml/5.0/rng/docbook.rng"
```

> Also the daps XSLT wrapper script under
>    /usr/share/daps/libexec/daps-xslt
> does not work on my openSUSE because it calls `/usr/bin/saxon` which
> does not exist.

Also not true - /usr/bin/saxon is part of the package saxon6-scripts which shoul be installed on openSUSE.

**Comment by tomschr** [comment]
**Commented by *anonymous on 2015-01-26 12:19**

> This is not true. DOCBOOK5_RNG_URI is defined in /etc/daps/config
>    DOCBOOK5_RNG_URI="http://docbook.org/xml/5.0/rng/docbook.rng"

Yes in fact, the variable is set in the global DAPS config file.

But the problem is that the command line calling jing in make/validate.mk of a DAPS installation uses the -c option which means RNC (RelaxNG compact) notation.
Therefor the configuration of a schema in RNG notation did not work for me and I wonder how this can work for anyone.

I still have to manually change it to

```
DOCBOOK5_RNG_URI="http://docbook.org/xml/5.0/rng/docbook.rnc"
```

to get my docbook5 sources validated.

**Comment by tomschr** [comment]
**Commented by fsundermeyer on 2015-01-26 12:32**
Thanks!
I just submitted a fix for it in trunk. Since the variable is name ..._RNG I chose to remove -c from the jing command line rather than changing the schema to RNC. I also added a note to the config file.

BTW: We started to seriously look into the DocBook5 support for DAPS and hope to have it ready in a few weeks. If that is done, we also will release DAPS 2.0.
